Code/Framework/AzCore/AzCore/DOM/Backends/JSON/JsonBackend.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/Backends/JSON/JsonSerializationUtils.h>
+#include <AzCore/DOM/DomBackend.h>
+
+namespace AZ::Dom
+{
+    //! A DOM backend for serializing and deserializing JSON <=> UTF-8 text
+    //! \param ParseFlags Controls how deserialized JSON is parsed.
+    //! \param WriteFormat Controls how serialized JSON is formatted.
+    template<
+        Json::ParseFlags ParseFlags = Json::ParseFlags::ParseComments,
+        Json::OutputFormatting WriteFormat = Json::OutputFormatting::PrettyPrintedJson>
+    class JsonBackend final : public Backend
+    {
+    public:
+        Visitor::Result ReadFromBuffer(const char* buffer, size_t size, AZ::Dom::Lifetime lifetime, Visitor& visitor) override
+        {
+            return Json::VisitSerializedJson<ParseFlags>({ buffer, size }, lifetime, visitor);
+        }
+
+        Visitor::Result ReadFromBufferInPlace(char* buffer, [[maybe_unused]] AZStd::optional<size_t> size, Visitor& visitor) override
```

Comment:
Not sure what the benefit is of adding the `AZStd::optional`. It just makes it harder to read and less efficient to process. Better approaches imho would be to just always require the size or have overloaded functions with and without the size. It's also inconsistent with the previous `ReadFromBuffer` call.

Reply:
It's optional because the string length might not (in the JSON case, definitely will not) be needed, so if we happen to want to process a string we don't know the length of, no additional work needs to be done.

Reply:
Sure, but `AZStd::optional` adds an additional variable to `size`, which, if used, means another branch here. Having 2 overloaded functions removes the need for such a check and imho makes the function more readable.